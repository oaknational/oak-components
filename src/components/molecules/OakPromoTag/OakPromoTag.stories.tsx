import React from "react";
import { Meta, StoryObj } from "@storybook/react";

import { OakPromoTag } from "./OakPromoTag";

const meta: Meta<typeof OakPromoTag> = {
  component: OakPromoTag,
  tags: ["autodocs"],
  title: "components/molecules/OakPromoTag",
};
export default meta;

type Story = StoryObj<typeof OakPromoTag>;

export const Default: Story = {
  render: () => <OakPromoTag />,
};
