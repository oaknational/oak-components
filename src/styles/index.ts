export * from "@/styles/theme/color";
