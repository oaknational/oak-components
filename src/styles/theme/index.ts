export { oakColorTokens, oakUiRoleTokens } from "./color";

export type {
  OakColorToken,
  OakCombinedColorToken,
  OakUiRoleToken,
  OakColorFilterToken,
} from "./color";

export {
  oakAllSpacingTokens,
  oakInnerPaddingTokens,
  oakSpaceBetweenTokens,
} from "./spacing";

export type {
  OakAllSpacingToken,
  OakCombinedSpacingToken,
  OakInnerPaddingToken,
  OakSpaceBetweenToken,
} from "./spacing";

export { oakDropShadowTokens } from "./dropShadow";
export type { OakDropShadowToken } from "./dropShadow";

export { oakOpacityTokens } from "./opacity";
export type { OakOpacityToken } from "./opacity";

export { oakTransitionTokens } from "./transitions";
export type { OakTransitionToken } from "./transitions";

export { oakBorderRadiusTokens, oakBorderWidthTokens } from "./borders";
export type { OakBorderRadiusToken, OakBorderWidthToken } from "./borders";

export type { OakTheme } from "./theme";

export { oakFontTokens, oakFontSizeTokens } from "./typography";
export type { OakFontToken, OakFontSizeToken } from "./typography";

export { oakZIndexTokens } from "./zIndex";

export { oakDefaultTheme } from "./default.theme";
