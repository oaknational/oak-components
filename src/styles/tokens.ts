export const oakAllSpacingPx = {
  "all-spacing-1": 4,
  "all-spacing-2": 8,
  "all-spacing-3": 12,
  "all-spacing-4": 16,
  "all-spacing-5": 20,
  "all-spacing-6": 24,
  "all-spacing-7": 32,
  "all-spacing-8": 40,
  "all-spacing-9": 48,
  "all-spacing-10": 56,
  "all-spacing-11": 64,
  "all-spacing-12": 72,
  "all-spacing-13": 80,
  "all-spacing-14": 92,
  "all-spacing-15": 100,
  "all-spacing-16": 120,
};

export type OakAllSpacing = keyof typeof oakAllSpacingPx;

export const oakInnerPaddingAllSpacing = {
  "inner-padding-xs": "all-spacing-2",
  "inner-padding-s": "all-spacing-3",
  "inner-padding-m": "all-spacing-4",
  "inner-padding-l": "all-spacing-5",
  "inner-padding-xl": "all-spacing-6",
};

export type OakInnerPadding = keyof typeof oakInnerPaddingAllSpacing;

export const oakSpaceBetweenAllSpacing = {
  "space-between-ssx": "all-spacing-2",
  "space-between-xs": "all-spacing-3",
  "space-between-s": "all-spacing-4",
  "space-between-m": "all-spacing-6",
  "space-between-m2": "all-spacing-7",
  "space-between-l": "all-spacing-9",
  "space-between-xl": "all-spacing-10",
  "space-between-xxl": "all-spacing-12",
  "space-between-xxxl": "all-spacing-13",
};

export type OakSpaceBetween = keyof typeof oakSpaceBetweenAllSpacing;

export const oakAllColorsHex = {
  white: "#ffffff",
  grey20: "#f2f2f2",
  grey30: "#e4e4e4",
  grey40: "#cacaca",
  grey50: "#808080",
  grey60: "#575757",
  grey70: "#2d2d2d",
  black: "#222222",
  oakGreen: "#287c34",
  mint: "bef2bd",
  mint30: "ebfbeb",
  mint50: "dff9de",
  aqua: "#b0e2de",
  aqua30: "#e7f6f5",
  aqua50: "#cee7e5",
  lavender: "#a0b6f2",
  lavender30: "#e3e9fb",
  lavender50: "#c6d1ef",
  pink: "#deb7d5",
  pink30: "#f5e9f2",
  pink50: "#e5d1e0",
  lemon: "#ffe555",
  lemon30: "#fff7cc",
  lemon50: "#f6e8a0",
  amber: "#ff934e",
  amber30: "#ffdfca",
  amber50: "#ffc8a6",
  red: "#dd0035",
  navy: "#0d24c4",
  blue: "#374cf1",
  magenta: "#d02aa7",
  purple: "#845ad9",
  teal: "#037b7d",
};

export type OakColor = keyof typeof oakAllColorsHex;

export const oakColorTokens = [
  "text-primary",
  "text-subdued",
  "text-error",
  "text-disabled",
  "text-link-active",
  "text-link-hover",
  "text-link-visited",
  "text-inverted",
  "text-success",
  "text-warning",
  "bg-primary",
  "bg-neutral",
  "bg-neutral-50",
  "bg-neutral-30",
  "bg-btn-primary",
  "bg-btn-primary-hover",
  "bg-btn-primary-disabled",
  "bg-btn-secondary",
  "bg-btn-secondary-hover",
  "bg-btn-secondary-disabled",
  "bg-icon",
  "bg-icon-hover",
  "bg-decorative1â€“main",
  "bg-decorative1-subdued",
  "bg-decorative1-very subdued",
  "bg-decorative2-main",
  "bg-decorative2-subdued",
  "bg-decorative2-very-subdued",
  "bg-decorative3",
  "bg-decorative3-50",
  "bg-decorative3-30",
  "bg-decorative4",
  "bg-decorative4-50",
  "bg-decorative4-30",
  "bg-decorative5",
  "bg-decorative5-50",
  "bg-decorative5-30",
  "bg-decorative6",
  "bg-decorative6-50",
  "bg-decorative6-30",
  "border-primary",
  "border-inverted",
  "border-neutral",
  "border-decorative1",
  "border-decorative2",
  "border-decorative3",
  "border-decorative4",
  "border-decorative5",
  "border-decorative6",
] as const;

export type OakColorToken = (typeof oakColorTokens)[number];
