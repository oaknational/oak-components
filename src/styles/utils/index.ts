export * from "./colorStyle";
export * from "./positionStyle";
export * from "./sizeStyle";
export * from "./spacingStyle";
export * from "./flexStyle";
