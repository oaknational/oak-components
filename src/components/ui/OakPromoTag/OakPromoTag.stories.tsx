import React from "react";
import { Meta, StoryObj } from "@storybook/react";

import { OakPromoTag } from "./OakPromoTag";

const meta: Meta<typeof OakPromoTag> = {
  component: OakPromoTag,
  tags: ["autodocs"],
  title: "components/ui/OakPromoTag",
};
export default meta;

type Story = StoryObj<typeof OakPromoTag>;

export const Default: Story = {
  render: () => <OakPromoTag />,
};
