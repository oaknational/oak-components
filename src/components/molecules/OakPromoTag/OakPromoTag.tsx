import React from "react";
import styled from "styled-components";

import { OakFlex, OakSpan } from "@/components/atoms";
import { InternalStyledSvg } from "@/components/atoms/InternalStyledSvg";
import { SizeStyleProps } from "@/styles/utils/sizeStyle";
import { DisplayStyleProps } from "@/styles/utils/displayStyle";

const StyledPromoTag = styled(OakFlex)`
  aspect-ratio: 2 / 1;
`;

export type OakPromoTagProps = {
  width?: SizeStyleProps["$width"];
  display?: DisplayStyleProps["$display"];
};

/**
 * Renders a tag with the text "New"
 */
export const OakPromoTag = (props: OakPromoTagProps) => {
  const {
    width = ["all-spacing-8", "all-spacing-9", "all-spacing-10"],
    display = "flex",
  } = props;

  return (
    <StyledPromoTag
      $alignItems="center"
      $justifyContent="center"
      $position="relative"
      $width={width}
      $display={display}
    >
      <InternalStyledSvg
        height="100%"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 56 27"
        $fill="bg-icon"
      >
        <path d="M6.421 0c8.661.839 17.303.834 26.16.714a89.843 89.843 0 0 1 4.834-.013c6.333.32 12.39.317 18.506-.275 0 1.26.178 2.352 0 3.44-.276 1.38-.829 2.759-1.342 4.15-.138.354-.532.706-.808 1.06-.445.248-.63.5-.553.753 1.598 1.093-.256 2.185 0 3.277.276 1.529 1.322 3.058 1.204 4.572-.138 1.771-1.046 3.543-1.973 5.312-.288.4-1.065.792-2.309 1.169-2.288.755-4.103 1.67-11.206 1.965-9.936.417-20.953.385-30.678-.09-2.447-.12-4.38-.532-5.209-.853-1.035-.458-1.52-.929-1.44-1.4C.937 18.988-.109 14.194.01 9.4c0-2.645 1.973-5.29 3.413-7.933.237-.424 1.697-.839 3-1.468Z" />
      </InternalStyledSvg>
      <OakFlex
        $position="absolute"
        $inset="all-spacing-0"
        $alignItems="center"
        $justifyContent="center"
      >
        <OakSpan
          $color="lemon"
          $font={["body-3-bold", "body-3-bold", "heading-7"]}
        >
          New
        </OakSpan>
      </OakFlex>
    </StyledPromoTag>
  );
};
