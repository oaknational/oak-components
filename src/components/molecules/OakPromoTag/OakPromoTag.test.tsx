import React from "react";
import "@testing-library/jest-dom";
import { create } from "react-test-renderer";
import { ThemeProvider } from "styled-components";

import { OakPromoTag } from "./OakPromoTag";

import { oakDefaultTheme } from "@/styles";

describe(OakPromoTag, () => {
  it("matches snapshot", () => {
    const tree = create(
      <ThemeProvider theme={oakDefaultTheme}>
        <OakPromoTag />
      </ThemeProvider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
