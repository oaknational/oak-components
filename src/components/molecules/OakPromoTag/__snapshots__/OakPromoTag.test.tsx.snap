// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakPromoTag matches snapshot 1`] = `
.c0 {
  position: relative;
  width: 2.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c4 {
  position: absolute;
  inset: 0rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  color: #ffe555;
  background: #222222;
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 700;
  font-size: 0.875rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: -0.005rem;
  -moz-letter-spacing: -0.005rem;
  -ms-letter-spacing: -0.005rem;
  letter-spacing: -0.005rem;
}

.c3 {
  fill: #222222;
}

.c2 {
  aspect-ratio: 2 / 1;
}

@media (min-width:750px) {
  .c0 {
    width: 3rem;
  }
}

@media (min-width:1280px) {
  .c0 {
    width: 3.5rem;
  }
}

@media (min-width:750px) {
  .c5 {
    font-weight: 700;
  }
}

@media (min-width:1280px) {
  .c5 {
    font-weight: 600;
  }
}

@media (min-width:750px) {
  .c5 {
    font-size: 0.875rem;
  }
}

@media (min-width:1280px) {
  .c5 {
    font-size: 1rem;
  }
}

@media (min-width:750px) {
  .c5 {
    line-height: 1.25rem;
  }
}

@media (min-width:1280px) {
  .c5 {
    line-height: 1.25rem;
  }
}

@media (min-width:750px) {
  .c5 {
    -webkit-letter-spacing: -0.005rem;
    -moz-letter-spacing: -0.005rem;
    -ms-letter-spacing: -0.005rem;
    letter-spacing: -0.005rem;
  }
}

@media (min-width:1280px) {
  .c5 {
    -webkit-letter-spacing: 0.0115rem;
    -moz-letter-spacing: 0.0115rem;
    -ms-letter-spacing: 0.0115rem;
    letter-spacing: 0.0115rem;
  }
}

<div
  className="c0 c1 c2"
>
  <svg
    className="c3"
    height="100%"
    viewBox="0 0 56 27"
    width="100%"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6.421 0c8.661.839 17.303.834 26.16.714a89.843 89.843 0 0 1 4.834-.013c6.333.32 12.39.317 18.506-.275 0 1.26.178 2.352 0 3.44-.276 1.38-.829 2.759-1.342 4.15-.138.354-.532.706-.808 1.06-.445.248-.63.5-.553.753 1.598 1.093-.256 2.185 0 3.277.276 1.529 1.322 3.058 1.204 4.572-.138 1.771-1.046 3.543-1.973 5.312-.288.4-1.065.792-2.309 1.169-2.288.755-4.103 1.67-11.206 1.965-9.936.417-20.953.385-30.678-.09-2.447-.12-4.38-.532-5.209-.853-1.035-.458-1.52-.929-1.44-1.4C.937 18.988-.109 14.194.01 9.4c0-2.645 1.973-5.29 3.413-7.933.237-.424 1.697-.839 3-1.468Z"
    />
  </svg>
  <div
    className="c4 c1"
  >
    <span
      className="c5"
    >
      New
    </span>
  </div>
</div>
`;
